### PR TITLE
feat: lint direct require quo component outside src/quo

### PR DESCRIPTION
**Pull request by yqrashawn**
### Summary

1. add script in `make lint` to lint against require quo component outside `src/quo`
1. quo2 -> quo in docs
1. ignore `/app` folder for prettier

error looks like 
CleanShot 2023-11-06 at 19 50 30

### Testing notes

dev exp only PR

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/wallet/accounts/common.cljs [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   (:require
-    [quo.components.markdown.text :as quo.text]
+    [quo.core :as quo]
     [quo.foundations.colors :as quo.colors]
     [re-frame.core :as re-frame]
     [react-native.core :as rn]
     [reagent.core :as reagent]
     [status-im.ui.components.chat-icon.screen :as chat-icon]
-    [status-im.ui.components.core :as quo]
+    [status-im.ui.components.core :as quo1]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/accounts/common.cljs]: better  `as components`

Comment (yqrashawn) [reply src/status_im/ui/screens/wallet/accounts/common.cljs]: fixed

File: nix/shells.nix [hunk 2]
```diff
@@ -21,7 +21,7 @@ let
       buildInputs = with pkgs; [
         clojure flock maven openjdk
         # lint specific utilities
-        clj-kondo zprint clojure-lsp
+        clj-kondo zprint clojure-lsp ripgrep
```
Comment (yqrashawn) [inline nix/shells.nix:24]: Hi @jakubgs, is it ok to put this here? Or into the default shell.

File: scripts/lint-direct-require-component-outside-quo.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env sh
+
+# quo components namespace ends with style|utils|types|reaction-resource
+# are not the component view, they are usually utils fns or styles
+
+if rg --pcre2 --glob '!src/quo/**/*' 'quo\.components(\.[\w-]+)*\.[\w-]*(?<!style|utils|types|reaction-resource)(\s|\]|\n)' src; then
```
Comment (J-Son89) [inline scripts/lint-direct-require-component-outside-quo.sh:6]: I'm not sure of these other file types, but if someone uses a `quo` component's `style.cljs` file outside of the quo namespace that should be a big code smell imo. I don't think it's something we need to lint for however 👍

**Review by J-Son89 — APPROVED** [review]
good stuff @yqrashawn! 💪

**Review by flexsurfer — APPROVED** [review]

File: scripts/lint-direct-require-component-outside-quo.sh (hunk 3 above)
Comment (yqrashawn) [reply scripts/lint-direct-require-component-outside-quo.sh:6]: These are usages I filterd out

CleanShot 2023-11-06 at 20 10 26

First 3 are used in old profile screen to pass new look into old components. I added these 3.
I guess we will remove these when new profile screen added.

2 of them are used in quo_preview. I'm not sure, maybe it's not required when using outside of preview

one is in comment

these 2 are indeed a bit weird
src/status_im2/contexts/emoji_picker/style.cljs
src/status_im2/contexts/chat/messages/drawers/view.cljs

Comment (J-Son89) [reply scripts/lint-direct-require-component-outside-quo.sh:6]: Oh let's lint this so, we really shouldn't have these uses imo

Comment (J-Son89) [reply scripts/lint-direct-require-component-outside-quo.sh:6]: Or at least create a follow up issue to address this

**Review by ilmotta — APPROVED** [review]
🏄🏼 🏄🏼‍♂️

File: nix/shells.nix (hunk 2 above)
Comment (jakubgs) [reply nix/shells.nix:24]: Yes, that's correct.

**Review by jakubgs — APPROVED** [review]

File: scripts/lint-direct-require-component-outside-quo.sh (hunk 3 above)
Comment (yqrashawn) [reply scripts/lint-direct-require-component-outside-quo.sh:6]: issue added
https://github.com/status-im/status-mobile/issues/17837 and https://github.com/status-im/status-mobile/issues/17838

**Pull request merged**